command/build.go

```diff
@@ -186,6 +189,8 @@ func (c *BuildCommand) RunContext(buildCtx context.Context, cla *BuildArgs) int
 	log.Printf("On error: %v", cla.OnError)
 
 	// Set the debug and force mode and prepare all the builds
+	// This is only affects json templates, because HCL2
+	// templates have already been prepared in GetBuilds() above.
```

Comment:
Yes that is true, a `CoreBuild` uses that field to make sure of it:

https://github.com/hashicorp/packer/blob/2981fd627d8ea7ebe5314c362a0f37e17f91e1e4/packer/build.go#L107